spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -645,10 +645,20 @@ else if (referencePersistentProperty.isCollectionLike()) {
 					Class descendantType = descendantPersistentProperty
 							.getComponentType();
 
+					Key entityKey = (Key) entity.getKey();
```

Comment:
Please refactor this into a helper method like `removeAncestorsFromKey(Key)`.

Reply:
Done.